### ISSUE TITLE
optional field causes error

### ISSUE THREAD

**Issue by Towerism** [mozilla/sphinx-js#75]
The error is:
```
Exception occurred:
  File "C:\Python27\lib\site-packages\sphinx_js\typedoc.py", line 123, in make_type_name
    node = self.nodelist[type.get('id')]
KeyError: 149
```
Caused by function with an optional argument like this:
```
function withOptional<T>(optional?: T) { /* ... */ }
```
Found a workaround:
```
function withOptional<T>(optional: T | undefined = undefined) { /* ... */ }

**Comment by WimYedema** [comment]
Ok, I'll do my best. But I won't be able to spend much time on it as I've just started in a new job.

**Comment by erikrose** [comment]
Hey, congratulations! Best of luck in the new gig.

**Comment by erikrose** [comment]
Fixed in the big #120 TypeScript rethink!
